kong/plugins/method-restriction/handler.lua

```diff
@@ -0,0 +1,32 @@
+local BasePlugin = require "kong.plugins.base_plugin"
+local responses = require "kong.tools.responses"
+local utils = require "kong.tools.utils"
+
+local MethodRestrictionHandler = BasePlugin:extend()
+
+MethodRestrictionHandler.PRIORITY = 940
+
+function MethodRestrictionHandler:new()
+  MethodRestrictionHandler.super.new(self, "method-restriction")
+end
+
+function MethodRestrictionHandler:access(conf)
+  MethodRestrictionHandler.super.access(self)
+
+  local block = false
+  local method = ngx.req.get_method()
```

Comment:
To avoid global lookups, `ngx.req.get_method` can be cached at the main chunk level of this module:
```lua
local get_method = ngx.req.get_method
```

And used as:
```lua
local method = get_method()
```